packages/terra-form/src/NumberField.jsx

```diff
@@ -107,6 +113,7 @@ const NumberField = (
     label={label}
     labelAttrs={labelAttrs}
     help={help}
+    htmlFor
```

Comment:
Should this be htmlFor={inputId}?

Missed setting id={inputId} for the Input component.